plugins/module_utils/network/sonic/config/logging/logging.py

```diff
@@ -0,0 +1,320 @@
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_logging_facts = self.get_logging_facts()
+
+        result['before'] = existing_logging_facts
+        if result['changed']:
+            result['after'] = changed_logging_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_logging_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        if want is None:
+            want = []
+
+        have = existing_logging_facts
+
+        resp = self.set_state(want, have)
+
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+
+        self.validate_want(want, state)
+        self.preprocess_want(want, state)
+
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+
+        return commands, requests
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :param want: the additive configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        diff = get_diff(want, have, TEST_KEYS)
+
+        commands = diff
+        requests = []
+        if commands:
+            requests = self.get_merge_requests(commands, have)
+
+        if len(requests) > 0:
+            commands = update_states(commands, "merged")
+        else:
+            commands = []
+
+        return commands, requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :param want: the objects from which the configuration should be removed
+        :param have: the current configuration as a dictionary
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        diff = get_diff(want, have, TEST_KEYS)
+
+        want_none = {'remote_servers': None}
```

Comment:
Discussed and clarified

Reply:
Mingjun has shown why this is needed. (Because of the default presence of {<key>; None} instances in Ansible parsed output, this implementation method must be used.)